src/proxy/outbound.rs

```diff
@@ -202,10 +207,43 @@ impl OutboundConnection {
                 self.proxy_to_tcp(source_stream, &req, &result_tracker)
                     .await
             }
+            Protocol::DOUBLEHBONE => {
+                self.proxy_to_double_hbone(source_stream, source_addr, &req, &result_tracker)
+                    .await
+            }
         };
         result_tracker.record(res)
     }
 
+    async fn proxy_to_double_hbone(
+        &mut self,
+        stream: TcpStream,
+        remote_addr: SocketAddr,
+        req: &Request,
+        connection_stats: &ConnectionResult,
+    ) -> Result<(), Error> {
+        let upgraded = Box::pin(self.send_hbone_request(remote_addr, req)).await?;
+
+        let upgraded = TokioH2Stream::new(upgraded);
+        let inner_workload = pool::WorkloadKey {
```

Comment:
Its unclear to me the exact mechanisms we are going to use to get info of e/w gateways and final destination.